enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DependencyEnricher.java

```diff
@@ -119,6 +120,12 @@ private void addArtifactsWithYaml(Set<URL> artifactSet, String dependencyYaml) {
 
     @Override
     public void adapt(final KubernetesListBuilder builder) {
+
+        getConfiguration().getResource().ifPresent(resourceConfig -> {
+            overrideEnvironmentVariables(builder, resourceConfig.getEnv()
+                .orElse(new HashMap<>()));
+        });
+
```

Comment:
Really not sure why we need to duplicate that piece of code in every enricher. For me that looks like a smell.

Whats about a post processing step outside of all enricher, which just overrides the environment variables added by the enricher chain ?